presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeJoinOptimizer.java

```diff
@@ -112,15 +114,35 @@ private boolean isMergeJoinEligible(PlanNode left, PlanNode right, JoinNode node
             // Acquire data properties for both left and right side
             StreamPropertyDerivations.StreamProperties leftProperties = StreamPropertyDerivations.derivePropertiesRecursively(left, metadata, session, types, parser);
             StreamPropertyDerivations.StreamProperties rightProperties = StreamPropertyDerivations.derivePropertiesRecursively(right, metadata, session, types, parser);
-
             List<VariableReferenceExpression> leftJoinColumns = node.getCriteria().stream().map(JoinNode.EquiJoinClause::getLeft).collect(toImmutableList());
```

Comment:
Let's also address Masha's comment in the merge join PR by adding comments about these assumptions to the merge join node:
https://github.com/facebookexternal/presto_cpp/pull/692

```
MergeJoinNode seems to be missing information about the sort order of the join keys. Is there an assumption that the sorting order is ASC and in case of multiple keys the order of the keys is the same as in "criteria"? 
```